### PR TITLE
Implement infrastructure for Nebula sub-site 🚀 

### PR DESCRIPTION
**Pull request by NPDebs**
Related to #765, #775, #841.

**[Deploy preview...](https://deploy-preview-867--ols-bebatut.netlify.app/nebula/about)**

**Screenshot:**
Screenshot 2024-08-10 at 03 21 15

**Changes made:**

- Adapted the `prepare_website_data.py` script to Nebula spreadsheet.
- Ran scripts to build basic infrastructure (of yaml files) and extract data from the planning sheet to populate them.
- Implemented "about page" design by Jilaga.
- Create conditional Nebula navbar design.

:+1::tada: First of all, thanks for taking the time to contribute! :tada::+1:

## FOR CONTRIBUTOR

* [x] I have read the [CONTRIBUTING.md](https://github.com/open-life-science/open-life-science.github.io/blob/main/CONTRIBUTING.md) document

PR Type: 
* [ ] This PR adds a new blog post
* [x] This PR does something else (explain above)

 
## FOR REVIEWERS

Thanks for taking the time to review! :heart:

Here are the list of things to make sure of:
* [ ] The website builds (a check will fail if not)
* [ ] All images have been added within the Pull Request and they have Alt text
* [ ] If there are paragraphs or text, the key messages are highlighted
* [ ] All internal links (within OLS website) use the [`{% link path_to_file.md %}` format](https://jekyllrb.com/docs/liquid/tags/#link)
* [ ] The preview corresponds to the changes described in the Pull Request
* [ ] The code is tidy and passes the linting tests

### CODE REVIEW

**Comment by iramosp** [comment]
Looking only at the general structure / navigation, it looks great! The navigation bar is how I imagined, and I especially love the landing page with @jilaga's design :)

A few navigation changes:
* **Clicking on the Nebula logo**
  * Current behavior: Points to `/nebula` (long program description)
  * Expected behavior: Points to `/nebula/about` (landing page). As a user, I'd expect the same behavior as clicking on `The Program` tab.
* **Clicking on the `Cohorts` tab**
  * Current behavior: Points to `/nebula/#cohorts`
  * Expected behavior: Either 1) do nothing, or 2) point to the current cohort, so `Nebula 2024-1` in the deploy preview, but soon to `Nebula 2024-2`. I don't have a preference, but I imagine doing nothing is one thing less we need to update for each new cohort.
* **Clicking on the `Resources` tab**
   * Current behavior: Does nothing, only displays dropdown menu
   * Expected behavior: Points to `/nebula/resources`. This tab doesn't need a dropdown menu because all resources will be listed in the same page.

**Comment by NPDebs** [comment]
Hi @iramosp. Thanks for the review! 🙏 

>* **Clicking on the Nebula logo**

If I understand you correctly, you want the logo to lead to the `/nebula/about`,  rather than `/nebula/` (i.e /nebula/index.md). That is very easily achievable. However, I'd like to call your attention to the fact that the `/about` page currently has almost the same info as /nebula.

**Comment by NPDebs** [comment]
>Clicking on the Cohorts tab

Both 1 and 2 sound like good options. Don't worry too much about having to change it every cohort - it's a small update that'll take only a few seconds.

The current structure is option 3 - point to the 'cohorts tables,' as seen below:
Screenshot 2024-08-09 at 03 09 19

It points to `/nebula/#cohorts` because ideally, we would like to have a table containing past cohorts at the bottom of `/nebula` page, no?

**Comment by iramosp** [comment]
> you want the logo to lead to the /nebula/about, rather than /nebula/ (i.e /nebula/index.md). 

* That's right, I see the deploy preview is already updated.


> the /about page currently has almost the same info as /nebula. What would be the difference between the landing page (/nebula/about) and /nebula?

* Once we have the structure, I will update the /nebula page with a more detailed program description, for example, the background for the program, roles, previous cohorts, and a detailed curriculum. The /about page will serve as a one-pager with only the necessary information for registration.


> Both 1 and 2 sound like good options.

* Ok, then let's do 2) Point to the current cohort. 


> It points to /nebula/#cohorts because ideally, we would like to have a table containing past cohorts at the bottom of /nebula page, no?

* Yes, we still want that. "Previous cohorts" from the dropdown menu already points there, and that's ok. My reasoning for the "Cohorts" tab pointing to the current cohort is to make it as simple as possible for participants to land on the right page for the current cohort, even if they don't read or explore other options from the dropdown menu. I know it's a tiny detail, but I think it makes navigation smoother.

Thanks for nudging me to explain these changes better.

**Comment by NPDebs** [comment]
> My reasoning for the "Cohorts" tab pointing to the current cohort is to make it as simple as possible for participants to land on the right page for the current cohort, even if they don't read or explore other options from the dropdown menu.

You have a really good sense of accessibility, @iramosp. Thanks a lot! 
Updates 1 & 2 have been made. Please, check that you're happy with them.

Also - the `resources` tab now points the resources page. Do you have a list of items you'd like to display here and a specific grouping pattern?

You can use the [policies-procedures-and-docs](https://we-are-ols.org/policies-procedures-and-docs/) page as a reference, thinking about the things that would benefit Nebula participants the most, and of course, preventing duplication.

**Comment by NPDebs** [comment]
> Once we have the structure, I will update the /nebula page with a more detailed program description...The /about page will serve as a one-pager with only the necessary information for registration.

@iramosp - I did a thing! Parts of the `about` page (especially those you'll be working with often) have been written in plain Markdown. The goal is to make it easier for you to tweak the file when you need to make small (or big!) changes.

The cohort schedule and timeline have also been added. Do let me know if we've migrated what we need from the index (`/nebula`) page to the `/about` page. 🙏 

@jilaga - How does it feel seeing your lovely design come to life? Do point out areas that you think could be implemented differently, please!

(_Also - Please do not feel obligated to respond outside your normal work hours_)

**Comment by yochannah** [comment]
soooooooooooo pretty! 

non tech review: can we have a little more side padding on the buttons please? 

screenshot of buttons with layout that is a little too cramped

**Comment by yochannah** [comment]
Two more requests (both discussed in our chat) 

- [ ] About doesn't link to `/about` right now in the navbar
- [ ] Apply basic styles from the about page to all nebula subpages 🙏  - particularly the heading style and the header image

**Comment by iramosp** [comment]
> check that you're happy with them

Yes, it looks good.

> Do you have a list of items you'd like to display here and a specific grouping pattern?

I have some things in mind. Could we get the structure ready first (ie this PR)? I think adding items to the resource page is just editing a md file (am I assuming right?), and I can do that in another PR so that this one doesn't become a bigger to-do list that we keep adding on.

> easier for you to tweak the file when you need to

Thank you! I do anticipate tweaking the `about` page often. 

Thanks @NPDebs!

**Comment by iramosp** [comment]
> About doesn't link to /about right now in the navbar

@yochannah, do you mean "About Nebula" in navbar points to /nebula/index.md instead of /about?

The logo and "The Program" in navbar point to the landing page (/about), while "About Nebula" in navbar points to /nebula/index.md. This is doing what I preferred in terms of content and navigation on the frontend, but I see the inconsistency with the file naming in the backend.  

Should we rename /about as /index.md and viceversa to keep consistency?

**Comment by yochannah** [comment]
I should have been clearer - the "about nebula" page doesn't seem to to anywhere! I've included a link of the screenshot and the URK it's trying to link to: 

image

**Comment by NPDebs** [comment]
> About doesn't link to /about right now in the navbar

@yochannah - This previously linked to an `application` section in the `index.md` page, but has now been linked properly (i.e `/nebula/index.md`).

> Should we rename /about as /index.md and viceversa to keep consistency?

This would be my recommendation, @iramosp, seeing as the homepage (A.K.A landing page) is named `index.md` by convention.

> Apply basic styles from the about page to all nebula subpages 🙏 - particularly the heading style and the header image

**Update:**
✅ Padding has been added to the buttons in the _Pricing_ section.

✅ As far as applying the basic styles in nebula subpages goes - I have used the same hero image everywhere for consistency. However, adding the styling to the headings would mean converting some or all of the pages from Markdown to HTML (to allow for these elements to be given _class names_ or _ids_ that have the style).

It is very doable, but the downside is that we'll be sacrificing "contributor user-friendliness" for aesthetics. Very happy to hear other ideas on how to get the best of both words.

**Comment by NPDebs** [comment]
> Could we get the structure ready first (ie this PR)? I think adding items to the resource page is just editing a md file (am I assuming right?), and I can do that in another PR so that this one doesn't become a bigger to-do list that we keep adding on.

Placing emphasis on the larger details in this PR, and leaving the bite-sized tasks for later would be my preference as well. 
And yes, you can create a PR using .md files, and we'll make the necessary, technical adjustments during the review. Sound good?

(P.S. I've created issue #872 for this purpose!)

**Comment by iramosp** [comment]
> you can create a PR using .md files, and we'll make the necessary, technical adjustments during the review. Sound good?

Perfect!

> we'll be sacrificing "contributor user-friendliness" for aesthetics

User-friendliness seems the priority here. Questions:

1. @yochannah, can we keep markdown with your proposal?
2. What does it mean applying the basic styles? I'm assuming it's just font and headings, but I'm not sure if it also refers to more complex elements like background colors
3. This is starting to feel like a different discussion than general structure and navigation. Could we open a new issue for this? I'm happy with the landing page and navigation, I think styling of other subpages can wait and probably needs input from Jilaga. 

@NPDebs, last few requests for this PR:
* Are we ready to go live? 
* Could you open another issue for styling? Again, we can leave this for later

**Comment by yochannah** [comment]
>     1. @yochannah, can we keep markdown with your proposal?

Yup! I agree with your other comments though - this shouldn't be a blocker.

File: _data/nebula/neb-1/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# List of experts, possible mentors and organizers for nebula neb-1
+#
+#
+# People should be also in people.yaml file and linked using their GitHub username
+# Ordering by expertise should be done by running the bin/sort-expertises.py script
+---
+experts: []
+organizers:
+- bebatut
```
Comment (yochannah) [inline _data/nebula/neb-1/metadata.yaml:9]:
```suggestion
- iramosp
- bebatut
```

**Review by yochannah — APPROVED** [review]
Approving based on the visuals, didn't do full code review.

**Comment by yochannah** [comment]
@bebatut Do you want me to open another issue for code review for this? I ask since you did part of it yourself already :)

**Comment by bebatut** [comment]
This PR did not pass the linting tests....

**Pull request merged**